elasticai_api/proto/elasticai_api.proto

```diff
@@ -24,6 +24,7 @@ message Shard {
   // Starting and ending (non-inclusive) record number.
   int64 start = 2;
   int64 end = 3;
+  repeated int64 indices = 4;
```

Comment:
I think we can make the shard small and then shuffle the shard list instead of shuffling each id and returning the id list.

Reply:
We usually store the images in the folder with the class name. For example:
```
|--mnist_png
    |--0
        |--001.png
        |--002.png
        |...
    |--1
       |--001.png
       |--002.png
```

Then, we can use `torchvision.datasets.ImageFolder("./mnist_png")` to load paths and classes of the images. 
![image](https://user-images.githubusercontent.com/18071380/103852276-8476a400-50e6-11eb-9137-4446133a761f.png)

If we create shards according to the `dataset.imgs`, the shard may only contains samples of a class. So we must shuffle all indices to create shards.